cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/statements/expressions/InitializerListExpression.kt

```diff
@@ -47,12 +47,7 @@ class InitializerListExpression : Expression(), ArgumentHolder, HasType.TypeObse
     /** The list of initializers. */
     @Relationship(value = "INITIALIZERS", direction = Relationship.Direction.OUTGOING)
     @AST
-    var initializerEdges = mutableListOf<PropertyEdge<Expression>>()
-        set(value) {
-            field.forEach { it.end.unregisterTypeObserver(this) }
-            field = value
-            value.forEach { it.end.registerTypeObserver(this) }
-        }
+    var initializerEdges = astEdgesOf<Expression>(postAdd = { it.end.registerTypeObserver(this) })
```

Comment:
And unregistering the TypeObserver happens automatically?

Reply:
No, still pending. The "remove" hook is not implemented yet.